lib/auth/auth.go

```diff
@@ -4200,13 +4207,48 @@ func (a *Server) UpsertDatabaseServer(ctx context.Context, server types.Database
 	return lease, nil
 }
 
+func (a *Server) UpdateOSSDesktopAlert(ctx context.Context) error {
+	windowsDesktops, err := a.GetWindowsDesktops(ctx, types.WindowsDesktopFilter{NonAD: []bool{true}})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	desktops := len(windowsDesktops)
+
+	if modules.GetModules().BuildType() == modules.BuildOSS && desktops > 3 {
+		alert, err := types.NewClusterAlert(OSSDesktopsAlertId, OSSDesktopsAlertMessage,
+			types.WithAlertSeverity(types.AlertSeverity_MEDIUM),
+			types.WithAlertLabel(types.AlertOnLogin, "yes"),
+			types.WithAlertLabel(types.AlertPermitAll, "yes"),
+			types.WithAlertExpires(time.Now().Add(200*24*365*time.Hour)))
```

Comment:
Is this just an arbitrary expiration far in the future? Can we just omit the expiry completely?

Reply:
It is, we have to provide something, if we don't it defaults to 24h

Reply:
I changed it to short-lived alerts and periodic checks